File: monitor/godaddy.py
"""GoDaddy (api.godaddy.com/v1). Header sso-key KEY:SECRET."""
from __future__ import annotations

from .base import http_get

API = "https://api.godaddy.com/v1"


def collect(account, token):
    headers = {"Authorization": f"sso-key {token}"}
    rows = []
    try:
        domains = http_get(f"{API}/domains", headers)
    except Exception as e:
        return [{
            "provider": "godaddy", "account": account, "service": "api_error",
            "resource_id": "domains", "name": "domains", "region": "",
            "status": "error", "cost_mtd": 0.0,
            "tags": {"error": str(e)[:200]}, "raw": {},
        }]
    for d in domains:
        rows.append({
            "provider": "godaddy", "account": account, "service": "domain",
            "resource_id": str(d.get("domainId", d.get("domain"))),
            "name": d.get("domain", ""), "region": "",
            "status": d.get("status", ""), "cost_mtd": 0.0,
            "tags": {"expires": str(d.get("expires", ""))},
            "raw": {"renewal": d.get("renewalPrice")},
        })
    return rows

File: dashboard/lib/monitor/providers/godaddy.ts
import { httpGet } from "../util";
import type { Row } from "../types";

const API = "https://api.godaddy.com/v1";

export async function collect(account: string, token: string): Promise<Row[]> {
  const h = { Authorization: `sso-key ${token}` };
  let domains: { domainId?: number; domain?: string; status?: string; expires?: string; renewalPrice?: number }[];
  try {
    domains = await httpGet(`${API}/domains`, h);
  } catch (e) {
    return [{
      provider: "godaddy", account, service: "api_error", resource_id: "domains",
      name: "domains", region: "", status: "error", cost_mtd: 0,
      tags: { error: String(e).slice(0, 200) }, raw: {},
    }];
  }
  return domains.map((d) => ({
    provider: "godaddy", account, service: "domain",
    resource_id: String(d.domainId ?? d.domain), name: d.domain || "",
    region: "", status: d.status || "", cost_mtd: 0,
    tags: { expires: String(d.expires || "") }, raw: { renewal: d.renewalPrice },
  }));
}
